handler/openid/helper.go

```diff
@@ -76,3 +89,34 @@ func (i *IDTokenHandleHelper) IssueExplicitIDToken(ctx context.Context, ar fosit
 	resp.SetExtra("id_token", token)
 	return nil
 }
+
+// ComputeHash computes the hash using the alg defined in the id_token header
+func (i *IDTokenHandleHelper) ComputeHash(ctx context.Context, sess Session, token string) (string, error) {
+	var err error
+	hashSize := 256
+	if alg, ok := sess.IDTokenHeaders().Get("alg").(string); ok {
+		hashSize, err = strconv.Atoi(alg[2:])
+		if err != nil {
+			hashSize = 256
+		}
+	}
+
+	var hash hash.Hash
+	if hashSize == 384 {
```

Comment:
I think it would make sense to set hash directly, without the intermediate hashSize variable :)

Reply:
Hmm, could you elaborate on this please? The `alg` may or may not be present in the header, so the first step here is to compute the hash size.

Reply:
I guess something along this pseudo-code:

```
hash := sha256.New()

if alg, ok := sess.IDTokenHeaders().Get("alg").(string); ok {
 		hashSize, _ := strconv.Atoi(alg[2:]) // 0 per default
 		switch hashSize {
 			case 384
				hash = sha512.New()
 			// ...
 		}
 	}

```

Reply:
@aeneasr Made the change.